pkg/runctl/controller.go

```diff
@@ -446,6 +481,7 @@ func (c *Controller) addPipelineRun(obj interface{}) {
 	}
 	klog.V(4).Infof("Add to workqueue '%s'", key)
 	c.workqueue.Add(key)
+	c.metricsWorkqueue.Add(key)
```

Comment:
I don't get that. For your metrics, you want get number for _all_ pipeline runs at regular intervals, not for those which are queued for processing by the controller. Otherwise you statistics will be garbage.